### PR TITLE
Handle DRPC deletion

### PR DESCRIPTION
**Pull request by BenamarMk**
When DRPC is deleted, the process will delete the ManifestWork for VRG, PV, and Roles
from the preferred cluster and the failover cluster.

### CODE REVIEW

File: controllers/util/mw_util.go [hunk 1]
```diff
@@ -489,20 +489,59 @@ func (mwu *MWUtil) createOrUpdateManifestWork(
...
+	}
+
+	err = mwu.deletePVManifestWork(clusterName)
+	if err != nil {
+		mwu.Log.Error(err, "failed to delete MW for PVs")
+
+		return fmt.Errorf("failed to delete ManifestWork for PVs in namespace %s (%w)", clusterName, err)
+	}
+
+	err = mwu.deleteRolesManifestWork(clusterName)
```
Comment (BenamarMk) [inline controllers/util/mw_util.go:507]: Note: We don't need to delete the roles MWs as @ShyamsundarR indicated.  The roles could be used by other apps.

File: controllers/drplacementcontrol_controller.go [hunk 2]
```diff
@@ -257,16 +285,105 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
...
 
 	return ctrl.Result{Requeue: true}, nil
 }
 
+func (r *DRPlacementControlReconciler) getDRPolicy(ctx context.Context,
+	name, namespace string) (*rmn.DRPolicy, error) {
+	drPolicy := &rmn.DRPolicy{}
+
+	err := r.Client.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, drPolicy)
+	if err != nil {
```
Comment (raghavendrabhat) [inline controllers/drplacementcontrol_controller.go:301]: (nit) Would logging the failure help in debugging?

File: controllers/drplacementcontrol_controller.go [hunk 3]
```diff
@@ -345,34 +466,45 @@ func (r *DRPlacementControlReconciler) annotatePlacementRule(ctx context.Context
...
-			clonedPlRule, err = r.clonePlacementRule(ctx, drpc, userPlRule, clonedPlRuleName)
+			clonedPlRule, err = r.clonePlacementRule(ctx, drPolicy, userPlRule, clonedPlRuleName)
 			if err != nil {
 				return nil, fmt.Errorf("failed to create cloned placementrule error: %w", err)
 			}
 		} else {
 			r.Log.Error(err, "Failed to get drpc placementRule", "name", clonedPlRuleName)
 
-			return nil, fmt.Errorf("failed to get placementrule error: %w", err)
+			return nil, err
```
Comment (raghavendrabhat) [inline controllers/drplacementcontrol_controller.go:485]: Would it be better to return by doing an explicit fmt.Errorf? The reason being IIUC the error message strings are to start with smaller case letters. In the above statement r.Log.Error will put a string that starts with capital letter to the err.

Let me know if I am missing something.

File: controllers/drplacementcontrol_controller.go (hunk 2 above)
Comment (BenamarMk) [reply controllers/drplacementcontrol_controller.go:301]: sure

File: controllers/drplacementcontrol_controller.go (hunk 3 above)
Comment (BenamarMk) [reply controllers/drplacementcontrol_controller.go:485]: I think the way it is is fine.  I am logging the error which should start with Capital letter.  And I am already wrapping the error (with lowercase) inside the getClonedPlacementRule.

**Review by ShyamsundarR — APPROVED** [review]
This commit to delete MWs and cleanup on DRPC deletion is fine.

We would need to ensure state of the system is as desired by DRPC when deleted, such that images and other entities are garbage collected right, for example is a half progressed failover is in progress, we may need to complete the same or back out appropriately. We can deal with this subsequently as we ensure restart-ability from any state in DRPC.

**Review by raghavendrabhat — APPROVED** [review]

**Pull request merged**